tools/delete.py

```diff
@@ -0,0 +1,134 @@
+                        help="Enter substring to be searched for in the data")
+    parser.add_argument("--endswith", metavar="endswith",
+                        help="Enter suffix to be searched for in the data")
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("firebase_credentials_file_url", metavar="firebase-credentials-file-url",
+                        help="GS URL to the private credentials file for the Firebase account where the phone "
+                             "number <-> uuid table is stored.")
+    parser.add_argument("firebase_table_name", metavar="firebase-table-name",
+                        help="Name of the data <-> uuid table in Firebase to use.")
+
+    args = parser.parse_args()
+
+    google_cloud_credentials_file_path = args.google_cloud_credentials_file_path
+    firebase_credentials_file_url = args.firebase_credentials_file_url
+    firebase_table_name = args.firebase_table_name
+    
+    prefix = args.startswith
+    substring = args.contains
+    suffix = args.endswith
+
+    log.info("Downloading Firestore UUID Table credentials...")
+    participants_uuid_table = init_uuid_table_client(google_cloud_credentials_file_path, firebase_credentials_file_url,
+                                                     firebase_table_name)
+
+    filter_by_prefix = lambda q: q[0].startswith(prefix)
+    filter_by_substring = lambda q: substring in q[0]
+    filter_by_suffix = lambda q: q[0].endswith(suffix)
+
+    firestore_filters = []
+    firestore_filter_desc = []
+
+    if prefix:
+        firestore_filters.append(filter_by_prefix)
+        firestore_filter_desc.append(f"start with {prefix}")
+
+    if substring:
+        firestore_filters.append(filter_by_substring)
+        firestore_filter_desc.append(f"contain {substring}")
+
+    if suffix:
+        firestore_filters.append(filter_by_suffix)
+        firestore_filter_desc.append(f"end with {suffix}")
+
+    if len(firestore_filters) == 0:
+        log.info("Filter mechanism not specified, skipping...")
+        exit(0)
+        
+    mappings = participants_uuid_table.get_all_mappings()
+    filtered_mappings = dict(filter( lambda x: all(f(x) for f in firestore_filters), mappings.items())) # As soon as a single filter returns False, that map element won't be included.
+
+    if not filtered_mappings:
+        log.info("No mappings to be deleted")
+        exit(0)
+        
+    log.warning(f"Deleting {len(filtered_mappings)} mapping(s) that {concat_description(firestore_filter_desc)}")
+    log.info("See sample data below ...")
+    limit = 5 if len(filtered_mappings) > 5 else len(filtered_mappings)
+    for i, k in zip(range(limit), filtered_mappings.keys()):
+        log.info(f"{k}")
```

Comment:
Deleting mappings is a very dangerous operation, so I think we should print all of them for verification